### PR TITLE
Implement eslint and improve prettier

### PR DESCRIPTION
**Pull request by vicjune**

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  printWidth: 100,
+  parser: 'flow',
+  tabWidth: 2,
+  singleQuote: true,
+  bracketSpacing: true,
+  arrowParens: 'avoid',
```
Comment (dmail) [inline .prettierrc.js]: In my experience `always` is nicer to work with because it's consistent.
You never have to add parenthesis because you add an argument or use destructuring.

```javascript
// let's say you use `avoid` configuration and omit parenthesis like below
const fn = a => a

// later you need to change it because it uses a second argument, you must add the parenthesis.
// prettier certainly made you lazy so you'll forgot and get a syntaxError (true story)
const fn = (a, b) => a + b

// same thing happens when you want to use destructuring
const fn = ({ name }) => name
```

My advice is to prefer `always`.

Comment (vicjune) [reply .prettierrc.js]: I agree with you on this, but then why did you chose on the player to remove semicolons ?

```javascript
if (
  extension.smth && extension.vmap ||
  extension.getExpression()
) {
  extension.doSomething();
  extension.doAnotherThing();
}
```
can be read in the blink of an eye, but you'll have to do a double take on this:
```javascript
if (
  extension.smth && extension.vmap ||
  extension.getExpression()
) {
  extension.doSomething()
  extension.doAnotherThing()
}
```

**Comment by eliamaino-fp** [comment]
We can rebase this

File: src/vmap.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { childrenByName, parseXMLNode } from './parser_utils';
 class VMAP {
   constructor(xml) {
     if (
-      (xml != null ? xml.documentElement : undefined) == null ||
+      (xml !== null ? xml.documentElement : undefined) === null ||
```
Comment (eliamaino-fp) [inline src/vmap.js]: I agree that this check was weak and error prone (legacy stuff from the coffeescript to es6 conversion), and that we should improve it. 

But this change doesn't do what we want, because if `xml` is `undefined` we try to access anyway `xml.documentElement` (since `undefined !== null = true`)

we should better change it to:
```javascript
if (
  !xml ||
  !xml.documentElement ||
  !xml.documentElement.localName !== 'VMAP'
) {
...
}
```

File: .prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  printWidth: 100,
+  parser: 'flow',
+  tabWidth: 2,
+  singleQuote: true,
+  bracketSpacing: true,
+  arrowParens: 'always',
+  trailingComma: 'es5',
+  semi: true,
+};
```
Comment (eliamaino-fp) [inline .prettierrc.js:10]: why did you move this file to `.js`?

Comment (vicjune) [reply .prettierrc.js:10]: To be consistent with .eslint.js and we can add comments and stuff

**Review by eliamaino-fp — APPROVED** [review]
